### DISCUSSION TITLE
FAST API ENDPOINTS respond with 404 NOT FOUND ERROR when run UVICORN ASGI SERVER from the same directory where APP instance located

### DISCUSSION THREAD

**Discussion by LovePelmeni** [fastapi/fastapi#6412] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python

# API.__init__.py 
from .controllers import controllers
# settings.py

from fastapi import FastAPI

application = fastapi.FastAPI()

# controllers.py 

from API import settings 
@application.post('/controller/')
def some_controller(request: fastapi.Request):
    return fastapi.Response(status_code=200)

# controllers/__init__.py 

from API import settings
```

### Description

Runs Fastapi application from "API" dir, endpoints return 404 
When runs from "FastAPIProject" dir, everyting works fine

### Operating System

macOS

### Operating System Details

MacOs Monterey 

### FastAPI Version

0.78.0

### Python Version

3.9

### Additional Context

I have a project with following simplified structure: 

        FastAPIProject - (Directory)
         |
         |
         |  ------API   (module)

                 |  -----__init__.py
                 |  -----settings.py

                 |  ---- controllers  (module)

                           |  -------__init__.py
                           |  -------controllers.py
         
 when I run `fastapi` from the `API` directory with `uvicorn settings:application --host 0.0.0.0 --port 8081` , so the postman responds with` 404`
`NOT FOUND` when i follow `http://0.0.0.0:8081/controller/` (the browser returns the same), 

but the Im in  `FASTAPIProject` directory  and do `uvicorn API.settings:application --host 0.0.0.0 --port 8081` the postman and browser responds with `200 OK`.

 Can I Fix this behaviour somehow without Moving Any Directories or modules? (Kinda hard for me because in reality it's pretty big app). 

Would really appreciate any help, probably I'm missing some details, but i've checked literally everything and could not find anything similar issues.

**Comment by LovePelmeni** [comment, marked as answer]
Problem is solved, looks like fastapi for some reasons does not see endpoints if it's been started from "module-typed" dir

**Comment by SheshaSri56** [comment]
hi,could you please explain the solution details..I am trying to solve the error...same 404(details not found error shows)

**Comment by LovePelmeni** [comment]
Hello, sorry if it's too late (been almost 2 years, since I've opened this issue). First, look at the diagram of directories I've created. If you have instance of FastAPI inside your `FastAPIProject/API/settings.py` file, in order to fix this problem with disappearing endpoints you have to go to `FastAPIProject/API` directory and run `uvicorn API.settings:application --port 8080` from there. Cannot explain why this is such a behaviour, but this is what I observed back then.
